packages/protocol/test/common/accounts.ts

```diff
@@ -0,0 +1,330 @@
+  const dataEncryptionKey = '0x02f2f48ee19680706196e2e339e5da3491186e0c4c5030670656b0e01611111111'
+  const longDataEncryptionKey =
+    '0x04f2f48ee19680706196e2e339e5da3491186e0c4c5030670656b0e01611111111' +
+    '02f2f48ee19680706196e2e339e5da3491186e0c4c5030670656b0e01611111111'
+
+  beforeEach(async () => {
+    accountsInstance = await Accounts.new({ from: account })
+    await accountsInstance.initialize()
+
+    authorizationTests.voting = {
+      fn: accountsInstance.authorizeVoteSigner,
+      eventName: 'VoteSignerAuthorized',
+      getAuthorizedFromAccount: accountsInstance.getVoteSignerFromAccount,
+      getAccountFromAuthorized: accountsInstance.getAccountFromVoteSigner,
+      getAccountFromActiveAuthorized: accountsInstance.getAccountFromActiveVoteSigner,
+    }
+    authorizationTests.validating = {
+      fn: accountsInstance.authorizeValidationSigner,
+      eventName: 'ValidationSignerAuthorized',
+      getAuthorizedFromAccount: accountsInstance.getValidationSignerFromAccount,
+      getAccountFromAuthorized: accountsInstance.getAccountFromValidationSigner,
+      getAccountFromActiveAuthorized: accountsInstance.getAccountFromActiveValidationSigner,
+    }
+    authorizationTests.attesting = {
+      fn: accountsInstance.authorizeAttestationSigner,
+      eventName: 'AttestationSignerAuthorized',
+      getAuthorizedFromAccount: accountsInstance.getAttestationSignerFromAccount,
+      getAccountFromAuthorized: accountsInstance.getAccountFromAttestationSigner,
+      getAccountFromActiveAuthorized: accountsInstance.getAccountFromActiveAttestationSigner,
+    }
+  })
+
+  describe('#initialize()', () => {
+    it('should set the owner', async () => {
+      const owner: string = await accountsInstance.owner()
+      assert.equal(owner, account)
+    })
+
+    it('should revert if already initialized', async () => {
+      await assertRevert(accountsInstance.initialize())
+    })
+  })
+
+  describe('#setAccountDataEncryptionKey()', () => {
+    it('should set dataEncryptionKey', async () => {
+      // @ts-ignore
+      await accountsInstance.setAccountDataEncryptionKey(dataEncryptionKey)
+      // @ts-ignore
+      const fetchedKey: string = await accountsInstance.getDataEncryptionKey(caller)
+      assert.equal(fetchedKey, dataEncryptionKey)
+    })
+
+    it('should allow setting a key with leading zeros', async () => {
+      const keyWithZeros = '0x00000000000000000000000000000000000000000000000f2f48ee19680706191111'
+      // @ts-ignore
+      await accountsInstance.setAccountDataEncryptionKey(keyWithZeros)
+      // @ts-ignore
+      const fetchedKey: string = await accountsInstance.getDataEncryptionKey(caller)
+      assert.equal(fetchedKey, keyWithZeros)
+    })
+
+    it('should revert when the key is invalid', async () => {
+      // @ts-ignore
+      await assertRevert(accountsInstance.setAccountDataEncryptionKey('0x32132931293'))
+    })
+
+    it('should allow a key that is longer than 33 bytes', async () => {
```

Comment:
What is special about 33?

Reply:
Not entirely sure, just copied it from `Attestations`. IIRC there was something about being able to register a full or compressed public key @trianglesphere ?